packages/fuels-programs/src/logs.rs

```diff
@@ -62,23 +63,81 @@ pub struct LogDecoder {
     pub log_formatters: HashMap<LogId, LogFormatter>,
 }
 
+#[derive(Debug)]
+pub struct LogResult {
+    pub results: Vec<Result<String>>,
+}
+
+impl LogResult {
+    pub fn filter_succeeded(&self) -> Vec<&str> {
+        self.results
+            .iter()
+            .filter_map(|result| result.as_deref().ok())
+            .collect()
+    }
+
+    pub fn filter_failed(&self) -> Vec<&Error> {
+        self.results
+            .iter()
+            .filter_map(|result| result.as_ref().err())
+            .collect()
+    }
+}
+
 impl LogDecoder {
     /// Get all decoded logs from the given receipts as `String`
-    pub fn get_logs(&self, receipts: &[Receipt]) -> Result<Vec<String>> {
-        receipts
+    pub fn decode_logs(&self, receipts: &[Receipt]) -> LogResult {
+        let results = receipts
             .iter()
             .extract_log_id_and_data()
-            .filter_map(|(log_id, data)| {
-                self.log_formatters
-                    .get(&log_id)
-                    .map(|log_formatter| log_formatter.format(&data))
+            .map(|(log_id, data)| self.format_log(&log_id, &data))
+            .collect();
+
+        LogResult { results }
+    }
+
+    fn format_log(&self, log_id: &LogId, data: &[u8]) -> Result<String> {
+        self.log_formatters
+            .get(log_id)
+            .ok_or_else(|| {
+                error!(InvalidData,"missing log formatter for log_id: `{:?}`. Consider adding external contracts with `set_contracts()`", log_id)
             })
-            .collect()
+            .and_then(|log_formatter| log_formatter.format(data))
+    }
+
+    pub fn decode_last_log(&self, receipts: &[Receipt]) -> Result<String> {
+        receipts
+            .iter()
+            .rev()
+            .extract_log_id_and_data()
+            .next()
+            .ok_or_else(|| error!(InvalidData, "nani"))
+            .and_then(|(log_id, data)| self.format_log(&log_id, &data))
+    }
+
+    pub fn decode_last_two_logs(&self, receipts: &[Receipt]) -> Result<(String, String)> {
```

Comment:
```suggestion 
fn decode_last_two_logs(&self, receipts: &[Receipt]) -> Result<(String, String)> {
```

This method doesn't need to be public I think.

Reply:
You're right.